validator_client/src/http_api/mod.rs

```diff
@@ -1254,14 +1251,43 @@ where
                     .body(vec![])
                     .expect("can produce simple response from static values"),
             };
-
-            let body: &Vec<u8> = response.body();
-            let signature = signer(body);
-            let header_value =
-                HeaderValue::from_str(&signature).expect("hash can be encoded as header");
-
-            response.headers_mut().append("Signature", header_value);
-
             response
-        })
+        }
+        Err(e) => {
+            let resp = warp_utils::reject::handle_rejection(e).await;
+            //Building the response with the Rejection
+            match resp {
+                Ok(reply) => {
+                    let response = reply.into_response();
+                    let (_parts, body) = response.into_parts();
+                    let body_bytes = hyper::body::to_bytes(body).await.unwrap();
```

Comment:
Avoid using `unwrap` as it could result in panic. If this conversion is needed and it fails then we could return an internal server error.